File: webstore/webstore-java/src/main/java/com/renan/webstore/business/AuthService.java
package com.renan.webstore.business;

import com.renan.webstore.controller.form.auth.*;
import com.renan.webstore.dto.UserDTO;

public interface AuthService {
    
    UserDTO login(LoginForm form) throws BusinessException;
    
    void recoverPassword(RecoverPasswordForm form) throws BusinessException;
    
    UserDTO resetPasswordByCode(PasswordResetForm form) throws BusinessException;
    
    void registerUser(UserRegistrationForm form) throws BusinessException;
    
    UserDTO activateUser(UserActivationForm form) throws BusinessException;
    
    void requestUserActivation(RequestUserActivationForm form) throws BusinessException;
    
}

File: webstore/webstore-kotlin/src/main/kotlin/com/renan/webstore/business/AuthService.kt
package com.renan.webstore.business

import com.renan.webstore.controller.auth.form.*
import com.renan.webstore.dto.UserDTO
import org.springframework.security.core.userdetails.UserDetailsService

interface AuthService: UserDetailsService {

    @Throws(BusinessException::class)
    fun login(form: LoginForm): UserDTO

    @Throws(BusinessException::class)
    fun recoverPassword(form: RecoverPasswordForm)

    @Throws(BusinessException::class)
    fun resetPasswordByCode(form: PasswordResetForm): UserDTO

    @Throws(BusinessException::class)
    fun registerUser(form: UserRegistrationForm)

    @Throws(BusinessException::class)
    fun activateUser(form: UserActivationForm): UserDTO

    @Throws(BusinessException::class)
    fun requestUserActivation(form: RequestUserActivationForm)

}